### PR TITLE
[#63] Add Terraform codebase

### PR DESCRIPTION
**Pull request by hoangmirs**
Closes https://github.com/nimblehq/infrastructure-templates/issues/63

## What happened 👀

- Move the general code into terraform add-on
- Add `.tool-versions` to support asdf
- Fix tests with the new change

## Insight 📝

- `main.tf`, `variables.tf`, `outputs.tf` are the general files, and they could be used by any providers, so we can move them into `terraform` add-on
- Revert `inquirer` to v8 as there are some importing errors 

## Proof Of Work 📹

- Tests should be passed
- Tested on my machine
index_ts_—_infrastructure-templates

### CODE REVIEW

**Comment by malparty** [comment]
I like the idea to isolate the `core` files of Terraform, but I'm quite confuse when we call this an `add-on`. By definition, an `add-on` is "something that has been or can be added to an existing object" (google). Though in our case, the existing object is the `terraform` core files, and we have the `aws` and `github` add-ons to add to the terraform core. 
So terraform files seems to be the core, not an add-on. 

In a similar idea, `add-ons` are IMO things we would ask for in the CLI command (`which provider do you use? which version control do you use? ...`).

**Review by malparty — APPROVED** [review]
LGTM now ;-)

**Comment by hoangmirs** [comment]
> I like the idea to isolate the `core` files of Terraform, but I'm quite confuse when we call this an `add-on`. By definition, an `add-on` is "something that has been or can be added to an existing object" (google). Though in our case, the existing object is the `terraform` core files, and we have the `aws` and `github` add-ons to add to the terraform core. So terraform files seems to be the core, not an add-on.
> 
> In a similar idea, `add-ons` are IMO things we would ask for in the CLI command (`which provider do you use? which version control do you use? ...`).

@malparty Thank you, you are right. It should be something like the core/general part.
I change it to core, and also create the core folder to put the code base. But if you have the better name/term, feel free to suggest it.

I also move `.gitignore` to core as it could be used either by GitLab or GitHub

File: skeleton/core/.tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+terraform 1.2.4
```
Comment (andyduong1920) [inline skeleton/core/.tool-versions:1]: just small comment should we use the latest version 1.2.8

**Review by andyduong1920 — APPROVED** [review]

**Review by andyduong1920 — APPROVED** [review]

**Pull request merged**